src/api-service/__app__/timer_retention/__init__.py

```diff
@@ -0,0 +1,85 @@
+import datetime
+import logging
+
+import azure.functions as func
+from onefuzztypes.enums import JobState, TaskState
+
+from ..onefuzzlib.events import get_events
+from ..onefuzzlib.jobs import Job
+from ..onefuzzlib.notifications.main import Notification
+from ..onefuzzlib.repro import Repro
+from ..onefuzzlib.tasks.main import Task
+
+RETENTION_POLICY = datetime.timedelta(days=(18 * 30))
+SEARCH_EXTENT = datetime.timedelta(days=(20 * 30))
+
+
+def main(mytimer1: func.TimerRequest, dashboard: func.Out[str]) -> None:  # noqa: F841
+
+    now = datetime.datetime.now(tz=datetime.timezone.utc)
+
+    time_retained_older = now - RETENTION_POLICY
+    time_retained_newer = now - SEARCH_EXTENT
+
+    time_filter = (
+        f"Timestamp lt datetime'{time_retained_older.isoformat()}' "
+        f"and Timestamp gt datetime'{time_retained_newer.isoformat()}'"
+    )
+    time_filter_newer = f"Timestamp gt datetime'{time_retained_older.isoformat()}'"
+
+    # Collecting 'still relevant' task containers.

```

Comment:
```suggestion
    # Collecting 'still relevant' task containers.
    # NOTE: This must be done before potentially modifying tasks otherwise
    # the task timestamps will not be useful.
```